pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -131,40 +159,37 @@ func ConvertIngressV1alpha3(ingress v1beta1.Ingress, domainSuffix string) (model
 			httpRoute.Match = []*networking.HTTPMatchRequest{httpMatch}
 			httpRoutes = append(httpRoutes, httpRoute)
 		}
-	}
-
-	if ingress.Spec.Backend != nil {
-		httpRoutes = append(httpRoutes, ingressBackendToHTTPRoute(ingress.Spec.Backend, ingress.Namespace, domainSuffix))
-	}
 
-	virtualService.Http = httpRoutes
+		virtualService.Http = httpRoutes
 
-	gatewayConfig := model.Config{
-		ConfigMeta: model.ConfigMeta{
-			Type:      model.Gateway.Type,
-			Group:     model.Gateway.Group,
-			Version:   model.Gateway.Version,
-			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
-			Namespace: model.IstioIngressNamespace,
-			Domain:    domainSuffix,
-		},
-		Spec: gateway,
-	}
+		virtualServiceConfig := model.Config{
+			ConfigMeta: model.ConfigMeta{
+				Type:      model.VirtualService.Type,
+				Group:     model.VirtualService.Group,
+				Version:   model.VirtualService.Version,
+				Name:      namePrefix + "-" + ingress.Name + "-" + model.IstioIngressGatewayName,
+				Namespace: model.IstioIngressNamespace,
+				Domain:    domainSuffix,
+			},
+			Spec: virtualService,
+		}
 
-	virtualServiceConfig := model.Config{
-		ConfigMeta: model.ConfigMeta{
-			Type:      model.VirtualService.Type,
-			Group:     model.VirtualService.Group,
-			Version:   model.VirtualService.Version,
-			Name:      ingress.Name + "-" + model.IstioIngressGatewayName,
-			Namespace: model.IstioIngressNamespace,
-			Domain:    domainSuffix,
-		},
-		Spec: virtualService,
+		old, f := ingressByHost[host]
```

Comment:
more idiomatic to call f ok. also, for ptrs, it's more conventional to test for zero value rather than using the bool ok value i.e. 
if old := ingressByHost[host]; old != nil {
see : https://blog.golang.org/go-maps-in-action